// @openrouter/sdk/esm/models/schema2.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: f5d5abd15e27
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const Schema4 = {
    Unknown: "unknown",
    OpenaiResponsesV1: "openai-responses-v1",
    AzureOpenaiResponsesV1: "azure-openai-responses-v1",
    XaiResponsesV1: "xai-responses-v1",
    AnthropicClaudeV1: "anthropic-claude-v1",
    GoogleGeminiV1: "google-gemini-v1",
};
/** @internal */
export const Schema4$inboundSchema = openEnums
    .inboundSchema(Schema4);
/** @internal */
export const Schema4$outboundSchema = openEnums
    .outboundSchema(Schema4);
/** @internal */
export const Schema2ReasoningText$inboundSchema = z.object({
    type: z.literal("reasoning.text"),
    text: z.nullable(z.string()).optional(),
    signature: z.nullable(z.string()).optional(),
    id: z.nullable(z.string()).optional(),
    format: z.nullable(Schema4$inboundSchema).optional(),
    index: z.number().optional(),
});
/** @internal */
export const Schema2ReasoningText$outboundSchema = z.object({
    type: z.literal("reasoning.text"),
    text: z.nullable(z.string()).optional(),
    signature: z.nullable(z.string()).optional(),
    id: z.nullable(z.string()).optional(),
    format: z.nullable(Schema4$outboundSchema).optional(),
    index: z.number().optional(),
});
export function schema2ReasoningTextToJSON(schema2ReasoningText) {
    return JSON.stringify(Schema2ReasoningText$outboundSchema.parse(schema2ReasoningText));
}
export function schema2ReasoningTextFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Schema2ReasoningText$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Schema2ReasoningText' from JSON`);
}
/** @internal */
export const Schema2ReasoningEncrypted$inboundSchema = z.object({
    type: z.literal("reasoning.encrypted"),
    data: z.string(),
    id: z.nullable(z.string()).optional(),
    format: z.nullable(Schema4$inboundSchema).optional(),
    index: z.number().optional(),
});
/** @internal */
export const Schema2ReasoningEncrypted$outboundSchema = z.object({
    type: z.literal("reasoning.encrypted"),
    data: z.string(),
    id: z.nullable(z.string()).optional(),
    format: z.nullable(Schema4$outboundSchema).optional(),
    index: z.number().optional(),
});
export function schema2ReasoningEncryptedToJSON(schema2ReasoningEncrypted) {
    return JSON.stringify(Schema2ReasoningEncrypted$outboundSchema.parse(schema2ReasoningEncrypted));
}
export function schema2ReasoningEncryptedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Schema2ReasoningEncrypted$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Schema2ReasoningEncrypted' from JSON`);
}
/** @internal */
export const Schema2ReasoningSummary$inboundSchema = z.object({
    type: z.literal("reasoning.summary"),
    summary: z.string(),
    id: z.nullable(z.string()).optional(),
    format: z.nullable(Schema4$inboundSchema).optional(),
    index: z.number().optional(),
});
/** @internal */
export const Schema2ReasoningSummary$outboundSchema = z.object({
    type: z.literal("reasoning.summary"),
    summary: z.string(),
    id: z.nullable(z.string()).optional(),
    format: z.nullable(Schema4$outboundSchema).optional(),
    index: z.number().optional(),
});
export function schema2ReasoningSummaryToJSON(schema2ReasoningSummary) {
    return JSON.stringify(Schema2ReasoningSummary$outboundSchema.parse(schema2ReasoningSummary));
}
export function schema2ReasoningSummaryFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Schema2ReasoningSummary$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Schema2ReasoningSummary' from JSON`);
}
/** @internal */
export const Schema2$inboundSchema = z.union([
    z.lazy(() => Schema2ReasoningSummary$inboundSchema),
    z.lazy(() => Schema2ReasoningEncrypted$inboundSchema),
    z.lazy(() => Schema2ReasoningText$inboundSchema),
]);
/** @internal */
export const Schema2$outboundSchema = z
    .union([
    z.lazy(() => Schema2ReasoningSummary$outboundSchema),
    z.lazy(() => Schema2ReasoningEncrypted$outboundSchema),
    z.lazy(() => Schema2ReasoningText$outboundSchema),
]);
export function schema2ToJSON(schema2) {
    return JSON.stringify(Schema2$outboundSchema.parse(schema2));
}
export function schema2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Schema2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Schema2' from JSON`);
}
